File: medplum-java/medplum-java-model/src/main/java/com/fhir/MedicationKnowledge_Regulatory.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationKnowledge_Regulatory.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationKnowledge_Regulatory {
  Reference regulatoryAuthority();

  Optional<MedicationKnowledge_MaxDispense> maxDispense();

  Optional<List<MedicationKnowledge_Substitution>> substitution();

  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<List<MedicationKnowledge_Schedule>> schedule();

  Optional<List<Extension>> modifierExtension();

  static ImmutableMedicationKnowledge_Regulatory.RegulatoryAuthorityBuildStage builder() {
    return ImmutableMedicationKnowledge_Regulatory.builder();
  }
}

File: medplum-typescript/src/model/fhir/MedicationKnowledge_Regulatory.ts
import { Extension } from "./Extension";
import { MedicationKnowledge_MaxDispense } from "./MedicationKnowledge_MaxDispense";
import { MedicationKnowledge_Schedule } from "./MedicationKnowledge_Schedule";
import { MedicationKnowledge_Substitution } from "./MedicationKnowledge_Substitution";
import { Reference } from "./Reference";

export interface MedicationKnowledge_Regulatory {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    regulatoryAuthority: Reference;
    substitution: MedicationKnowledge_Substitution[] | null | undefined;
    schedule: MedicationKnowledge_Schedule[] | null | undefined;
    maxDispense: MedicationKnowledge_MaxDispense | null | undefined;
}
